assertj-error-prone/src/main/java/com/palantir/assertj/errorprone/AssertjSize.java

```diff
@@ -0,0 +1,143 @@
+            .named("isEqualTo");
+    private static final Matcher<ExpressionTree> isLessThan = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isLessThan");
+    private static final Matcher<ExpressionTree> isLessThanOrEqualTo = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isLessThanOrEqualTo");
+    private static final Matcher<ExpressionTree> isGreaterThan = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isGreaterThan");
+    private static final Matcher<ExpressionTree> isGreaterThanOrEqualTo = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isGreaterThanOrEqualTo");
+    private static final Matcher<ExpressionTree> isZero = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isZero")
+            .withParameters();
+    private static final Matcher<ExpressionTree> isNotZero = MethodMatchers.instanceMethod()
+            .onDescendantOf("org.assertj.core.api.Assert")
+            .namedAnyOf("isNotZero")
+            .withParameters();
+    private static final Matcher<ExpressionTree> comparisonTo =
+            Matchers.anyOf(isEqualTo, isLessThan, isLessThanOrEqualTo, isGreaterThan, isGreaterThanOrEqualTo);
+
+    private static final Matcher<ExpressionTree> globalComparison = Matchers.anyOf(isZero, isNotZero);
+
+    private final AssertjSingleAssertMatcher matcher = AssertjSingleAssertMatcher.of(this::match);
+
+    @Override
+    public Optional<AssertjCheckerResult> matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        int arguments = tree.getArguments().size();
+        if ((arguments == 1 && comparisonTo.matches(tree, state))
+                || (arguments == 0 && globalComparison.matches(tree, state))) {
+            return matcher.matches(tree, state);
+        }
+        return Optional.empty();
+    }
+
+    private Optional<AssertjCheckerResult> match(
+            AssertjSingleAssertMatcher.SingleAssertMatch match, VisitorState state) {
+        ExpressionTree tree = ASTHelpers.stripParentheses(match.getAssertThat().getArguments().get(0));
+        if (!sizeMatcher.matches(tree, state)) {
+            return Optional.empty();
+        }
+        String replacementName = sizeComparisonName(match.getCheck(), state);
+        return Optional.of(AssertjCheckerResult.builder()
+                .description(DESCRIPTION)
+                .fix(SuggestedFix.builder()
+                        .replace(
+                                tree, state.getSourceForNode(ASTHelpers.getReceiver(ASTHelpers.stripParentheses(tree))))
+                        .merge(SuggestedFixes.renameMethodInvocation(match.getCheck(), replacementName, state))
```

Comment:
Equality matchers are broader thann size matchers, this needs to check if the result type can be assigned to `int`, otherwise cast.